packages/llamaindex/src/vector-store/PGVectorStore.ts

```diff
@@ -131,10 +136,19 @@ export class PGVectorStore
       }
     }
 
-    const db = this.db;
+    if (this.db && !this.dbConnected) {
+      try {
+        await this.db.connect();
+      } catch (error) {
+        console.error(error);
+        return Promise.reject(
```

Comment:
im personally don't like try-catch block to check if already connected. do we have other solution?

Reply:
> personally don't like try-catch block to check if already connected

I wouldn't it call it that. The 
```ts
if (this.db && !this.dbConnected) {}
```

in combination with the following event listener

```ts
db.on("end", () => (this.dbConnected = false));
```

Acts as a "re-connection" handler.

With this change,  whenever you call `getDb()` or `client()` it will try to return a client that's already connected to the DB.

If you prefer we can remove this logic completely

Reply:
At the end of the day I just want to have the ability to customize my PG client (e.g, pass certs) and to configure the PGVectorStore's `dimensions` and etc.

So please give a guide on what's the way that you want to go